src/sentry/static/sentry/app/components/events/interfaces/spans/spanDetail.tsx

```diff
@@ -3,86 +3,257 @@ import styled from 'react-emotion';
+import EventView from 'app/views/eventsV2/eventView';
+import {generateDiscoverResultsRoute} from 'app/views/eventsV2/results';
 
 import {SpanType} from './types';
 
-type PropTypes = {
+type TransactionResult = {
+  'project.name': string;
+  transaction: string;
+  id: string;
+};
+
+type Props = {
+  api: Client;
+  orgId: string;
   span: Readonly<SpanType>;
+  isRoot: boolean;
+  eventView: EventView;
 };
 
-const SpanDetail = (props: PropTypes) => {
-  const {span} = props;
+type State = {
+  transactionResults?: TransactionResult[];
+};
 
-  const startTimestamp: number = span.start_timestamp;
-  const endTimestamp: number = span.timestamp;
+class SpanDetail extends React.Component<Props, State> {
+  state: State = {
+    transactionResults: undefined,
+  };
 
-  const duration = (endTimestamp - startTimestamp) * 1000;
-  const durationString = `${duration.toFixed(3)} ms`;
+  componentDidMount() {
+    const {span} = this.props;
 
-  return (
-    <SpanDetailContainer
-      data-component="span-detail"
-      onClick={event => {
-        // prevent toggling the span detail
-        event.stopPropagation();
-      }}
-    >
-      <table className="table key-value">
-        <tbody>
-          <Row title="Span ID">{span.span_id}</Row>
-          <Row title="Trace ID">{span.trace_id}</Row>
-          <Row title="Parent Span ID">{span.parent_span_id || ''}</Row>
-          <Row title="Description">{get(span, 'description', '')}</Row>
-          <Row title="Start Date">
-            <React.Fragment>
-              <DateTime date={startTimestamp * 1000} />
-              {` (${startTimestamp})`}
-            </React.Fragment>
-          </Row>
-          <Row title="End Date">
-            <React.Fragment>
-              <DateTime date={endTimestamp * 1000} />
-              {` (${endTimestamp})`}
-            </React.Fragment>
-          </Row>
-          <Row title="Duration">{durationString}</Row>
-          <Row title="Operation">{span.op || ''}</Row>
-          <Row title="Same Process as Parent">
-            {String(!!span.same_process_as_parent)}
-          </Row>
-          <Tags span={span} />
-          {map(get(span, 'data', {}), (value, key) => {
-            return (
-              <Row title={key} key={key}>
-                {JSON.stringify(value, null, 4) || ''}
-              </Row>
-            );
-          })}
-          <Row title="Raw">{JSON.stringify(span, null, 4)}</Row>
-        </tbody>
-      </table>
-    </SpanDetailContainer>
-  );
-};
+    this.fetchSpanDescendents(span.span_id)
+      .then(response => {
+        if (
+          !response.data ||
+          !Array.isArray(response.data) ||
+          response.data.length <= 0
+        ) {
+          return;
+        }
+
+        this.setState({
+          transactionResults: response.data,
+        });
+      })
+      .catch(_error => {
+        // don't do anything
+      });
+  }
+
+  fetchSpanDescendents(spanID: string): Promise<any> {
```

Comment:
Could move this into an actioncreator module. Having components doing both request/response handing and rendering DOM gets hard to maintain.

Reply:
I'll do this in a follow up PR.